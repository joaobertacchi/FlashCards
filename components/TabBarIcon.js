// @flow

import React from 'react';
import { Icon } from 'expo';

import Colors from '../constants/Colors';

type Props = {
  focused: boolean,
  name: string,
  type?: string,
};

const TabBarIcon = (props: Props) => {
  const { focused, name, type = 'Ionicons' } = props;
  const iconSet = {
    AntDesign: Icon.AntDesign,
    Entypo: Icon.Entypo,
    EvilIcons: Icon.EvilIcons,
    Feather: Icon.Feather,
    FontAwesome: Icon.FontAwesome,
    Foundation: Icon.Foundation,
    Ionicons: Icon.Ionicons,
    MaterialIcons: Icon.MaterialIcons,
    MaterialCommunityIcons: Icon.MaterialCommunityIcons,
    SimpleLineIcons: Icon.SimpleLineIcons,
    Octicons: Icon.Octicons,
    Zocial: Icon.Zocial,
  };
  const MyIcon = iconSet[type];
  return (
    <MyIcon
      name={name}
      size={26}
      style={{ marginBottom: -3 }}
      color={focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
};

TabBarIcon.defaultProps = {
  type: 'Ionicons',
};

export default TabBarIcon;
