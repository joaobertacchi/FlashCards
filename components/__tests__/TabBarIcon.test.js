import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import TabBarIcon from '../TabBarIcon';

describe('TabBarIcon snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer.create(<TabBarIcon name="md-add" focused={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer.create(<TabBarIcon name="md-add" focused />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer
      .create(<TabBarIcon name="cards-outline" focused={false} type="MaterialCommunityIcons" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer
      .create(<TabBarIcon name="cards-outline" focused type="MaterialCommunityIcons" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
