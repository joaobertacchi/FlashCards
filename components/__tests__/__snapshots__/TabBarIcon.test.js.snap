// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon snapshot renders correctly Next Question button 1`] = `<Text />`;

exports[`TabBarIcon snapshot renders correctly Next Question button 2`] = `<Text />`;

exports[`TabBarIcon snapshot renders correctly Next Question button 3`] = `<Text />`;

exports[`TabBarIcon snapshot renders correctly Next Question button 4`] = `<Text />`;
